simplestore/src/main/java/com/uber/simplestore/impl/SimpleStoreImpl.java

```diff
@@ -129,8 +132,8 @@
     requireOpen();
     return Futures.submitAsync(
         () -> {
-          if (isTombstoned()) {
-            return Futures.immediateFailedFuture(new StoreClosedException());
+          if (isDead() != null) {
```

Comment:
Is this a race condition? If so, save isDead return value to a field.

Reply:
Nope, there's an underlying atomicint

Reply:
I meant that we call this method on separate lines but assume that it keeps the same value.